v2/binding/transformer/extract_metadata_test.go

```diff
@@ -0,0 +1,124 @@
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/stretchr/testify/assert"
+)
+
+func ExampleExtractExtensions() {
+	e := event.New("1.0")
+	e.Context = &event.EventContextV1{}
+	e.SetID("id")
+	e.SetSource("source")
+	e.SetType("type")
+	e.SetExtension("someextension", "somevalue")
+	e.SetExtension("otherextension", "othervalue") // extension will be ignored
+	extractor := ExtractExtensions{"someextension": nil}
+	_, err := binding.ToEvent(context.TODO(), binding.ToMessage(&e), extractor)
+	if err != nil {
+		fmt.Printf("Error running extractor: %v", err)
+	}
+	fmt.Println("extractor:", extractor)
+	//Output:
+	//extractor: map[someextension:somevalue]
+}
+
+func TestExtractExtensions(t *testing.T) {
+	t.Parallel()
+
+	e := test.MinEvent()
+	e.Context = e.Context.AsV1()
+	e.SetExtension("extension1", "extension1-val")
+	e.SetExtension("extension2", false)
+
+	tcs := []struct {
+		name      string
+		event     *event.Event
+		extractor ExtractExtensions
+		want      ExtractExtensions
+	}{
+		{
+			name:  "no extensions extracted",
+			event: &e,
+			extractor: ExtractExtensions{
+				"extension3": nil,
+			},
+			want: ExtractExtensions{
+				"extension3": nil,
+			},
+		},
+		{
+			name:  "one extension extracted",
+			event: &e,
+			extractor: ExtractExtensions{
+				"extension3": nil,
+				"extension1": nil,
+			},
+			want: ExtractExtensions{
+				"extension1": "extension1-val",
+				"extension3": nil,
+			},
+		},
+		{
+			name:  "two extensions extracted",
+			event: &e,
+			extractor: ExtractExtensions{
+				"extension1": nil,
+				"extension2": nil,
+				"extension3": nil,
+			},
+			want: ExtractExtensions{
+				"extension1": "extension1-val",
+				"extension2": false,
+				"extension3": nil,
+			},
+		},
+	}
+	for _, tc := range tcs {
+		eventExtractor := make(ExtractExtensions)
+		binaryExtractor := make(ExtractExtensions)
+		structuredExtractor := make(ExtractExtensions)
+		for k, v := range tc.extractor {
+			eventExtractor[k] = v
+			binaryExtractor[k] = v
+			structuredExtractor[k] = v
+		}
+		testArgs := []test.TransformerTestArgs{
+			{
+				Name:         "event",
+				InputEvent:   tc.event.Clone(),
+				AssertFunc:   assertExtractExtensions(tc.event, eventExtractor, tc.want),
+				Transformers: []binding.TransformerFactory{eventExtractor},
+			},
+			{
+				Name:         "binary",
+				InputMessage: test.MustCreateMockBinaryMessage(*tc.event),
+				AssertFunc:   assertExtractExtensions(tc.event, binaryExtractor, tc.want),
+				Transformers: []binding.TransformerFactory{binaryExtractor},
+			},
+			{
+				Name:         "structured",
+				InputMessage: test.MustCreateMockStructuredMessage(*tc.event),
+				AssertFunc:   assertExtractExtensions(tc.event, structuredExtractor, tc.want),
+				Transformers: []binding.TransformerFactory{structuredExtractor},
+			},
+		}
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			test.RunTransformerTests(t, context.TODO(), testArgs)
```

Comment:
Just a small comment about test organization: can you put this `RunTransformerTests` outside the loop? prepare all testArgs, then just invoke the run test one time without wrapping `t.Run` again. This makes clearer the view of the test cases in IDEs

Reply:
I don't use an IDE which is bothered by this so I'm not sure what the difference is, but I like having the different variations grouped as subtests of the same test case.

Reply:
ok fine